packages/dds/sequence/src/test/overlappingIntervalsIndex.spec.ts

```diff
@@ -0,0 +1,460 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable jsdoc/check-indentation */
+
+import { strict as assert } from "assert";
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils";
+import { LocalReferencePosition, compareReferencePositions } from "@fluidframework/merge-tree";
+import { IRandom, makeRandom } from "@fluid-internal/stochastic-test-utils";
+import { sequenceIntervalHelpers, IntervalType, SequenceInterval } from "../intervalCollection";
+import { SharedString } from "../sharedString";
+import { SharedStringFactory } from "../sequenceFactory";
+import { sequenceIntervalIndexFactory } from "../intervalIndex";
+import { RandomIntervalOptions } from "./intervalIndexUtils";
+
+class TestSharedString extends SharedString {
+	// Expose the `client` to the public and keep other properties unchanged
+	public client;
+}
+
+function assertSequenceIntervalsEqual(
+	string: TestSharedString,
+	results: SequenceInterval[],
+	expected: { start: number; end: number }[] | SequenceInterval[],
+): void {
+	assert.equal(results.length, expected.length, "Mismatched result count");
+
+	for (let i = 0; i < results.length; ++i) {
+		assert(results[i]);
+		const resultStart = string.localReferencePositionToPosition(results[i].start);
+		const resultEnd = string.localReferencePositionToPosition(results[i].end);
+		let expectedStart;
+		let expectedEnd;
+
+		if (expected[i] instanceof SequenceInterval) {
+			expectedStart = string.localReferencePositionToPosition(
+				expected[i].start as LocalReferencePosition,
+			);
+			expectedEnd = string.localReferencePositionToPosition(
+				expected[i].end as LocalReferencePosition,
+			);
+		} else {
+			expectedStart = expected[i].start;
+			expectedEnd = expected[i].end;
+		}
+
+		assert.equal(resultStart, expectedStart, "mismatched start");
+		assert.equal(resultEnd, expectedEnd, "mismatched end");
+	}
+}
+
+/**
+ * Generates a random string of length 2 or 1, where each character is chosen from the range 'a' to 'z'.
+ */
+function genreateRandomText(random: IRandom) {
+	const length = random.bool(0.5) ? 2 : 1;
+	let result = "";
+	const characters = "abcdefghijklmnopqrstuvwxyz";
+
+	for (let i = 0; i < length; i++) {
+		result += characters[random.integer(0, 25)];
+	}
+
+	return result;
```

Comment:
```suggestion
	return random.string(random.bool() ? 2 : 1);
```

It doesn't have exactly the same behavior, but `random.string()` already only uses pretty readable characters: "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ". I'm guessing this is fine for your purposes.